File: frontend/src/app/_shared/models/pessoa/paciente/paciente.ts
import { Endereco } from "../endereco";
import { Pessoa } from "../pessoa";
import { Contato } from "./contato";

export class Paciente extends Pessoa {
    constructor(
        public peso?: number,
        public imc?: number,
        public altura?: number,
        public socioeconomico?: string,
        public escolaridade?: string,
        public contatos?: Contato[],
        id?: number,
        nome?: string,
        sexo?: string,
        idade?: number,
        endereco?: Endereco,
        dataNasc?: Date,
        cpf?: string,
        telefone?: string
    ) {
        super(id, nome, sexo, idade, endereco, dataNasc, cpf, telefone);
    }
}

File: frontend/mobile/app/src/main/java/com/tecno/aging/domain/models/pessoa/paciente/Paciente.kt
package com.tecno.aging.domain.models.pessoa.paciente

data class PacienteEndereco(
    val logradouro: String?,
    val numero: Int?,
    val bairro: String?,
    val municipio: String?,
    val uf: String?,
    val cep: String?,
    val complemento: String?
)

data class Paciente(
    val id: Int,
    val nome: String,
    val cpf: String,
    val idade: Int,
    val peso: Double?,
    val altura: Double?,
    val imc: Double?,
    val socioeconomico: String?,
    val escolaridade: String?,
    val estadoCivil: String?,
    val nacionalidade: String?,
    val municipioNasc: String?,
    val ufNasc: String?,
    val corRaca: String?,
    val rg: String?,
    val dataExpedicao: String?,
    val orgaoEmissor: String?,
    val ufEmissor: String?,
    val dataNasc: String?,
    val telefone: String?,
    val sexo: String?,
    val fotoPerfil: String? = null,
    val endereco: PacienteEndereco?
)